### PR TITLE
Highlighted navigation drawer item

### LINKED ISSUE #163: Highlighted item  in Navigation Drawer according to the currently selected destination
## Kind (Required)

- Improvement

 ## Overview (Required)
Now the Home item is always highlighted in Navigation Drawer.

How about making it the item is highlighted according to the currently selected destination?
For example, this is what the DroidKaigi2020 app looks like.

https://user-images.githubusercontent.com/19250035/109412185-e3231480-79e9-11eb-9ca3-761ee52bb425.mp4

 ## Links	

- https://developer.android.com/jetpack/compose/navigation?hl=ja#bottom-nav

### PR DESCRIPTION
**Pull request by Dai1678**
## Issue
- close #163 

## Overview (Required)
- Highlighted drawer item when you clicked drawer item

## What I won't do in this PR
- Highlighted drawer item when you clicked **Tab item**

Sorry, I couldn't think of a better solution...

## Links
- FYI: https://developer.android.com/jetpack/compose/navigation#bottom-nav

## Screenshot
Before | After
:--: | :--:
 |

### CODE REVIEW

**Comment by Dai1678** [comment]
> What I won't do in this PR
Highlighted drawer item when you clicked Tab item

I'm sorry. I couldn't think of a good solution.
Does anyone have a good solution?
Or please help me as another welcome contribute issue!

**Comment by takahirom** [comment]
👀

File: uicomponent-compose/main/src/main/java/io/github/droidkaigi/feeder/AppContent.kt [hunk 1]
```diff
@@ -46,7 +48,13 @@ fun AppContent(
         drawerState = drawerState,
         drawerShape = MaterialTheme.shapes.large.copy(all = CornerSize(0.dp)),
         drawerContent = {
-            DrawerContent { route ->
+            val navBackStackEntry by navController.currentBackStackEntryAsState()
```
Comment (takahirom) [inline uicomponent-compose/main/src/main/java/io/github/droidkaigi/feeder/AppContent.kt:51]: How about this implementation? 👀 

```kotlin
 var currentDestination by remember { mutableStateOf<NavDestination>()}
    navController.addOnDestinationChangedListener { controller, destination: NavDestination, arguments ->
        currentDestination = destination
    }
```

Comment (takahirom) [reply uicomponent-compose/main/src/main/java/io/github/droidkaigi/feeder/AppContent.kt:51]: I will check it a little more. 👀

Comment (takahirom) [reply uicomponent-compose/main/src/main/java/io/github/droidkaigi/feeder/AppContent.kt:51]: If you do this implementation (addOnDestinationChangedListener), a lot of listeners will be added, so you have to remember or do something. 👀

**Comment by takahirom** [comment]
Currently, I think this is the best implementation 👍 
By the way, do you think it should be reflected when tabs are selected?

**Comment by Dai1678** [comment]
I think what complicates this issue is that the Drawer and Tab have the same role.
However, I think it would be difficult to change the design drastically, so maybe we don't need to do the highlighting item in Drawer.

Quit using `DrawerButton.isSelected`

I'm interested in hearing other people's opinions.

**Comment by takahirom** [comment]
Well, it may be better to create an issue. Can you make it if you like?

**Comment by Dai1678** [comment]
Okay. I will created a new issue.
Thank you so much!

**Comment by Dai1678** [comment]
@takahirom 
I created a new issue #180 

Will this PR be merged?

**Comment by takahirom** [comment]
OK! Thank you for thinking about this!

**Review by takahirom — APPROVED** [review]

**Pull request merged**